vulpes/upgrades/gamestate.cpp

```diff
@@ -401,6 +406,10 @@ void init_gamestate_upgrades() {
     patch_copy_checkpoint_file_hook.apply();
     gamestate_copy_checkpoint_file_continue_ptr = patch_copy_checkpoint_file_hook.return_address();
 
+    // Almost double the amount of memory objects have to work with.
+    patch_increase_object_memory_pool.build(sig_object_memory_pool_size());
```

Comment:
I want to make sure I understand this correctly -- is `sig_object_memory_pool_size()` performing a scan for the address matching what you've defined in YAML? Then this patch holds on to the original contents of that address so you can revert it later?

Would you ever `apply()` a patch but not `build()` it, or vice-versa? If they are always done together, I think a combined method like `apply(sig_object_memory_pool_size())` would ensure users of a `Patch` don't forget to call both `build()` and `apply()` and in the correct order.

Reply:
sig is actually getting the address from a previously done scan.

Patches have to be built because that way the original contents and new contents can be generated because it all happens on the fly.

Calling patch before build actually causes an assertion error.

I don't want to build stuff on patch application because there is good reasons to keep them separate. At times we need to build a patch at a different time than we apply it.

And some patches are allowed to fail building too. (But they are not allowed to be applied if that happens.)

Build actually outputs a boolean about if it was successful. The only true failure condition of that function is if the address it was given isn't valid.

The reason the boolean isn't checked here is because this signature is crucial. Simply put, if it wasn't found then this code wouldn't be executing.